kore/app/rpc/Main.hs

```diff
@@ -2,8 +2,172 @@
 
 module Main (main) where
 
+import Control.Monad.Catch (
+    bracket,
+    handle,
+ )
+import Control.Monad.Reader (
+    ReaderT (..),
+ )
+import GlobalMain qualified
+import Kore.BugReport (
+    BugReportOption,
+    ExitCode,
+    parseBugReportOption,
+    withBugReport,
+ )
+import Kore.Exec (
+    SerializedModule (..),
+ )
 import Kore.JsonRpc (runServer)
+import Kore.Log (
+    KoreLogOptions (..),
+    parseKoreLogOptions,
+    runKoreLog,
+ )
+import Kore.Log.ErrorException (
+    handleSomeException,
+ )
+import Kore.Rewrite.SMT.Lemma (declareSMTLemmas)
+import Kore.Syntax.Definition (
+    ModuleName (..),
+ )
+import Log qualified
+import Options.Applicative (
+    InfoMod,
+    Parser,
+    argument,
+    auto,
+    fullDesc,
+    header,
+    help,
+    long,
+    metavar,
+    option,
+    str,
+ )
+import Options.SMT (
+    KoreSolverOptions (..),
+    ensureSmtPreludeExists,
+    parseKoreSolverOptions,
+ )
 import Prelude.Kore
+import SMT qualified
+import System.Clock (
+    Clock (Monotonic),
+    TimeSpec,
+    getTime,
+ )
+import System.Exit (
+    ExitCode (ExitSuccess),
+    exitWith,
+ )
+
+data KoreRpcServerOptions = KoreRpcServerOptions
+    { definitionFileName :: !FilePath
+    , mainModuleName :: !ModuleName
+    , koreSolverOptions :: !KoreSolverOptions
+    , koreLogOptions :: !KoreLogOptions
+    , bugReportOption :: !BugReportOption
+    , port :: !Int
+    }
+
+-- | Command Line Argument Parser
+parseKoreRpcServerOptions :: TimeSpec -> Parser KoreRpcServerOptions
+parseKoreRpcServerOptions startTime =
+    KoreRpcServerOptions
+        <$> argument
+            str
+            ( metavar "DEFINITION_FILE"
+                <> help "Kore definition file to verify and use for execution"
+            )
+        <*> parseMainModuleName
+        <*> parseKoreSolverOptions
+        <*> parseKoreLogOptions Main.exeName startTime
+        <*> parseBugReportOption
+        <*> option
+            auto
```

Comment:
not sure what that range is tbh.

Reply:
`[1 .. 65535]` (but using [1..1024] require `root` privileges). This is very minor, though, illegal port numbers will crash when starting inside `runrerver` otherwise

Reply:
yeah, it might also crash if the port is taken